pages/firebase-firsttry.js

```diff
@@ -0,0 +1,38 @@
+import { useEffect, useState } from 'react';
+import { collection, onSnapshot } from 'firebase/firestore';
+import Link from 'next/link';
+
+import { db } from '../firebase/initFirebase';
+
+const FirebaseFirst = () => {
+  const [articleList, setArticleList] = useState([]);
+  const articlesCollectionRef = collection(db, 'articles');
+
+  useEffect(() => {
+    onSnapshot(articlesCollectionRef, (snapshot) => {
+      setArticleList(
+        snapshot.docs.map((doc) => ({
+          ...doc.data(),
+          id: doc.id,
+        })),
+      );
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  return (
+    <div className="homepage">
+      <h1 className="headline--1 mb-4">Kategorie</h1>
+      {articleList.map((article) => (
+        <div key={article.id}>
+          <Link href={`/article/${article.link}`}>
+            <a className="underline text-blue-600">
+              <h2>{article.name}</h2>
+            </a>
+          </Link>
+        </div>
```

Comment:
You could also do this. No strong opinion here. Do as you like

```suggestion
      {articleList.map(({id, link, name}) => (
        <div key={id}>
          <Link href={`/article/${link}`}>
            <a className="underline text-blue-600">
              <h2>{name}</h2>
            </a>
          </Link>
        </div>
```

Reply:
ok :D